guides/v2.1/howdoi/checkout/checkout-add-custom-carrier.md

```diff
@@ -0,0 +1,297 @@
+                    <frontend_class>shipping-applicable-country</frontend_class>
+                    <source_model>Magento\Shipping\Model\Config\Source\Allspecificcountries</source_model>
+                </field>
+                <field id="specificcountry" translate="label" type="multiselect" sortOrder="70" showInDefault="1" showInWebsite="1" showInStore="0">
+                    <label>Ship to Specific Countries</label>
+                    <source_model>Magento\Directory\Model\Config\Source\Country</source_model>
+                    <can_be_empty>1</can_be_empty>
+                </field>
+                <field id="showmethod" translate="label" type="select" sortOrder="80" showInDefault="1" showInWebsite="1" showInStore="0">
+                    <label>Show Method if Not Applicable</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <frontend_class>shipping-skip-hide</frontend_class>
+                </field>
+                <field id="sort_order" translate="label" type="text" sortOrder="90" showInDefault="1" showInWebsite="1" showInStore="0">
+                    <label>Sort Order</label>
+                </field>
+            </group>
+        </section>
+    </system>
+</config>
+```
+
+### Source code of `app/code/Vendor/CustomShipping/etc/config.xml`
+
+File specifies default values for custom shipping module options and shipping module model - `Vendor\CustomShipping\Model\Carrier\Customshipping`.
+
+```xml
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Store:etc/config.xsd">
+    <default>
+        <carriers>
+            <customshipping>
+                <active>0</active>
+                <title>Custom Shipping Title</title>
+                <name>Custom Shipping Method Name</name>
+                <shipping_cost>10</shipping_cost>
+                <sallowspecific>0</sallowspecific>
+                <sort_order>999</sort_order>
+                <model>Vendor\CustomShipping\Model\Carrier\Customshipping</model>
+            </customshipping>
+        </carriers>
+    </default>
+</config>
+```
+
+## Step 3: Create configuration file {#carrier-model}
+
+### Source code of `app/code/Vendor/CustomShipping/Model/Carrier/Customshipping.php`
+
+```php
+<?php
+
+namespace Vendor\CustomShipping\Model\Carrier;
+
+use Magento\Quote\Model\Quote\Address\RateRequest;
+use Magento\Shipping\Model\Carrier\AbstractCarrier;
+use Magento\Shipping\Model\Carrier\CarrierInterface;
+
+/**
+ * Custom shipping model
+ */
+class Customshipping extends AbstractCarrier implements CarrierInterface
+{
+    /**
+     * @var string
+     */
+    protected $_code = 'customshipping';
+
+    /**
+     * @var bool
+     */
+    protected $_isFixed = true;
+
+    /**
+     * @var \Magento\Shipping\Model\Rate\ResultFactory
+     */
+    protected $_rateResultFactory;
```

Comment:
The `$_code` and `$_isFixed` are inherited from the parent class. `$_rateResultFactory`and `$_rateMethodFactory` are introduced within the scope of the current class. So, let's follow the best practices wherever we can by removing the underscore from `$_rateResultFactory`and `$_rateMethodFactory` and changing scope to private.

Reply:
**implemented**

Thank you for advise! 👍